services/httpd/handler_test.go

```diff
@@ -318,6 +318,49 @@ func TestHandler_Ping(t *testing.T) {
 	}
 }
 
+// Ensure the handler returns the version correctly from the different endpoints.
+func TestHandler_Version(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	tests := []struct {
+		method   string
+		endpoint string
+		body     *bytes.Reader
+	}{
+		{
+			method:   "GET",
+			endpoint: "/ping",
+			body:     nil,
+		},
+		{
+			method:   "GET",
+			endpoint: "/query?db=foo&q=SELECT+*+FROM+bar",
+			body:     nil,
+		},
+		{
+			method:   "POST",
+			endpoint: "/write",
+			body:     bytes.NewReader(make([]byte, 10)),
+		},
+	}
+
+	for _, test := range tests {
+		if test.body != nil {
```

Comment:
@e-dard That's what I thought too, but it returns this:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0x124dcb]

goroutine 29 [running]:
testing.func·006()
        /Users/ross/.gvm/gos/go1.4.3/src/testing/testing.go:441 +0x181
net/http.NewRequest(0x5414a0, 0x3, 0x53dc80, 0x5, 0xc13938, 0x0, 0xa, 0x0, 0x0)
        /Users/ross/.gvm/gos/go1.4.3/src/net/http/request.go:516 +0x45b
github.com/influxdata/influxdb/services/httpd_test.MustNewRequest(0x5414a0, 0x3, 0x53dc80, 0x5, 0xc13938, 0x0, 0xc2080c0f40)
        /Users/ross/gocodez/src/github.com/influxdata/influxdb/services/httpd/handler_test.go:544 +0x66
github.com/influxdata/influxdb/services/httpd_test.TestHandler_Version(0xc2080657a0)
        /Users/ross/gocodez/src/github.com/influxdata/influxdb/services/httpd/handler_test.go:348 +0x329
testing.tRunner(0xc2080657a0, 0x7e5c10)
        /Users/ross/.gvm/gos/go1.4.3/src/testing/testing.go:447 +0xbf
created by testing.RunTests
        /Users/ross/.gvm/gos/go1.4.3/src/testing/testing.go:555 +0xa8b
```

Which I'm assuming has to do with the `body` member being a `*bytes.Reader`, so it's getting a `nil` pointer instead of just a `nil` value. I tried a few other things, but the way it is now is the only way I could make it work.